SoomlaAndroidProfile/src/com/soomla/profile/SoomlaProfile.java

```diff
@@ -1152,6 +1158,170 @@ public void multiShare(String text, String imageFilePath) {
+            @Override
+            public void success(List<Leaderboard> result, boolean hasMore) {
+                if (reward != null) {
+                    reward.give();
+                }
+                BusProvider.getInstance().post(new GetLeaderboardsFinishedEvent(provider, result, payload));
+            }
+
+            @Override
+            public void fail(String message) {
+                BusProvider.getInstance().post(new GetLeaderboardsFailedEvent(provider, message, payload));
+            }
+        });
+    }
+
+    /**
+     Fetches the game's scores list from specified leaderboard
+
+     * @param provider The provider to use
+     * @param leaderboard Leaderboard containing desired scores list
+     * @param reward   The reward to grant
+     * @throws ProviderNotFoundException if the supplied provider is not
+     *                                   supported by the framework
+     */
+    public void getScores(final IProvider.Provider provider, final Leaderboard leaderboard, final Reward reward) throws ProviderNotFoundException {
+        getScores(provider, leaderboard, false, "", reward);
+    }
+
+    /**
+     Fetches the game's scores list from specified leaderboard
+
+     * @param provider The provider to use
+     * @param leaderboard Leaderboard containing desired scores list
+     * @param fromStart Should we reset pagination or request the next page
+     * @param reward   The reward to grant
+     * @throws ProviderNotFoundException if the supplied provider is not
+     *                                   supported by the framework
+     */
+    public void getScores(final IProvider.Provider provider, final Leaderboard leaderboard, final boolean fromStart, final Reward reward) throws ProviderNotFoundException {
+        getScores(provider, leaderboard, fromStart, "", reward);
+    }
+
+    /**
+     Fetches the game's scores list from specified leaderboard
+
+     * @param provider The provider to use
+     * @param leaderboard Leaderboard containing desired scores list
+     * @param fromStart Should we reset pagination or request the next page
+     * @param payload  a String to receive when the function returns.
+     * @param reward   The reward to grant
+     * @throws ProviderNotFoundException if the supplied provider is not
+     *                                   supported by the framework
+     */
+    public void getScores(final IProvider.Provider provider, final Leaderboard leaderboard, final boolean fromStart, final String payload, final Reward reward) throws ProviderNotFoundException {
+        final IGameServicesProvider gsProvider = mProviderManager.getGameServicesProvider(provider);
+
+        BusProvider.getInstance().post(new GetScoresStartedEvent(provider, leaderboard, fromStart, payload));
+        gsProvider.getScores(leaderboard.getID(), fromStart, new GameServicesCallbacks.SuccessWithListListener<Score>() {
```

Comment:
@EugeneButusov, sorry I missed this before...
`getID()` breaks naming conventions in Java, change it to `getId()`